inductiva/resources/machine_types.py

```diff
@@ -0,0 +1,11 @@
+"""Available machine types and their number of cores."""
+AVAILABLE_MACHINES = {
+    "c2-standard-": [4, 8, 16, 30, 60],
+    "c3-standard-": [4, 8, 22, 44, 88, 176],
+    "c2d-standard-": [2, 4, 8, 16, 32, 56, 112],
+    "c2d-highcpu-": [2, 4, 8, 16, 32, 56, 112],
+    "e2-standard-": [2, 4, 8, 16, 32],
+    "n2-standard-": [2, 4, 8, 16, 32, 48, 64, 80, 96, 128],
+    "n2d-standard-": [2, 4, 8, 16, 32, 48, 64, 80, 96, 128, 224],
+    "n1-standard-": [1, 2, 4, 8, 16, 32, 64, 96],
+}
```

Comment:
add an utility method here that returns a generator of all available machines:

```python
def list_available_machines():
    return (machine_type+str(vcpu)
            for machine_type, vcpus in AVAILABLE_MACHINES.items()
            for vcpu in vcpus)
```

 It can then be used elsewhere. Ex:

```python
if 'xpto-standard-3' not in list_available_machines():
    raise Exception()
```

The gist is that we can then replace the body of the `list_available_machines` with an API call whenever the backend implements that endpoint , with minor changes to the overall code.